### ISSUE TITLE
bug: CloudPod of a Lambda function with hot-reloading contain absolute path

### ISSUE THREAD

**Issue by joe4dev** [localstack/localstack#9974]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Creating a CloudPod of a Lambda function with [hot-reloading](https://docs.localstack.cloud/user-guide/tools/lambda-tools/hot-reloading/) encodes an absolute path. Hence, the CloudPod is not cross-machine compatible and loading the CloudPod on another machine leads to such errors:

Function invocation:

```log
2023-11-29 16:14:43 2023-11-29T15:14:43.796 ERROR --- [   asgi_gw_8] l.s.lambda_.provider       : Error while invoking lambda
2023-11-29 16:14:43 Traceback (most recent call last):
2023-11-29 16:14:43   File "/opt/code/localstack/.venv/lib/python3.11/site-packages/docker/api/client.py", line 268, in _raise_for_status
2023-11-29 16:14:43     response.raise_for_status()
2023-11-29 16:14:43   File "/opt/code/localstack/.venv/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
2023-11-29 16:14:43     raise HTTPError(http_error_msg, response=self)
2023-11-29 16:14:43 requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/5f77f01ced1e56cdd2b5fbc9863b73390501f00034f0d7c0debf70a71a564383/start
```

Docker mount denied error:

> 2023-11-29 16:14:43 docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.41/containers/5f77f01ced1e56cdd2b5fbc9863b73390501f00034f0d7c0debf70a71a564383/start: Internal Server Error ("b'Mounts denied: \nThe path /home/runner/work/localstack-platform/localstack-platform is not shared from the host and is not known to Docker.\nYou can configure shared paths from Docker -> Preferences... -> Resources -> File Sharing.\nSee https://docs.docker.com/desktop/mac for more info.'")

### Expected Behavior

Options:

a) Convert any hot-reloaded Lambda function (i.e., `HotReloadingCode(ArchiveCode)`) into a regular S3-backed Lambda function (i.e., `S3Code(ArchiveCode)`) upon persisting.
b) Provide some path mapping when loading a hot-reloaded Lambda function. This is not very user-friendly if users just want a persisted Lambda function and do not care about hot-reloading upon loading the CloudPod.

### How are you starting LocalStack?

Custom (please describe below)

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

`localstack start`

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

1. Create a Lambda function with [hot-reloading](https://docs.localstack.cloud/user-guide/tools/lambda-tools/hot-reloading/)
2. Create a CloudPod: `localstack pod save hot-lambda`
3. Inspect the CloudPod and discover a hard-coded absolute path to the mounted Lambda code or try to load it on another machine

### Environment

```markdown
- OS: macOS
- LocalStack: latest (2023-12)
```

### Anything else?

Internal discussion in [#sig-platform](https://localstack-cloud.slack.com/archives/C02ETJ7PD63/p1701271035831489)

**Comment by giograno** [comment]
This issue has been solved by @dfangl with the workaround in https://github.com/localstack/localstack/pull/10857.
[Persistence docs](https://docs.localstack.cloud/user-guide/state-management/support/) point to hot-reloading lambda docs for this particular use case.

**Comment by joe4dev** [comment]
> This issue has been solved by @dfangl with the workaround in #10857. [Persistence docs](https://docs.localstack.cloud/user-guide/state-management/support/) point to hot-reloading lambda docs for this particular use case.

Thank you 👍 Nice example.

The docs are a bit hard to find without a link (I guess markdown links are not supported) and "that section" could be ambiguous. Maybe adding the page and/or section title could clarify.
